src/code/z_kankyo.c

```diff
@@ -667,27 +980,289 @@ Color_RGBA8 sSkyboxEnvColors[] = {
+        // From here, 0x10 is decremented from nextDayTime until it reaches either 0xFF0E or 0xFF0D, effectively
+        // delaying the chicken crow or dog howl sfx by 15 frames when loading the new area.
+
+        if (gSaveContext.nextDayTime == (NEXT_TIME_DAY_SET - (15 * 0x10))) {
+            Audio_PlaySfx(NA_SE_EV_CHICKEN_CRY_M);
+            gSaveContext.nextDayTime = NEXT_TIME_NONE;
+        } else if (gSaveContext.nextDayTime == (NEXT_TIME_NIGHT_SET - (15 * 0x10))) {
+            Audio_PlaySfx_2(NA_SE_EV_DOG_CRY_EVENING);
+            gSaveContext.nextDayTime = NEXT_TIME_NONE;
+        }
+    }
+}
+
+void Environment_UpdateTime(PlayState* play, EnvironmentContext* envCtx, PauseContext* pauseCtx, MessageContext* msgCtx,
+                            GameOverContext* gameOverCtx) {
+    u16 time;
+
+    if (!sEnvIsTimeStopped && (pauseCtx->state == PAUSE_STATE_OFF) && (gameOverCtx->state == GAMEOVER_INACTIVE)) {
+        if ((msgCtx->msgMode == MSGMODE_NONE) || (msgCtx->currentTextId == 0xF7) || (msgCtx->currentTextId == 0x20D2) ||
+            (msgCtx->currentTextId == 0x140C) ||
+            ((msgCtx->currentTextId >= 0x100) && (msgCtx->currentTextId <= 0x200)) ||
+            (gSaveContext.gameMode == GAMEMODE_END_CREDITS)) {
+            if (!FrameAdvance_IsEnabled(&play->state) &&
+                ((play->transitionMode == TRANS_MODE_OFF) || (gSaveContext.gameMode != GAMEMODE_NORMAL))) {
+                if (play->transitionTrigger == TRANS_TRIGGER_OFF) {
+                    if ((CutsceneManager_GetCurrentCsId() == CS_ID_NONE) && !Play_InCsMode(play)) {
+                        gSaveContext.save.time = ((void)0, gSaveContext.save.time) + (u16)R_TIME_SPEED;
+                        if (R_TIME_SPEED != 0) {
+                            gSaveContext.save.time =
+                                ((void)0, gSaveContext.save.time) + (u16)((void)0, gSaveContext.save.timeSpeedOffset);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if ((((void)0, gSaveContext.skyboxTime) >= CLOCK_TIME(6, 0)) ||
+        (((void)0, gSaveContext.save.time) < CLOCK_TIME(6, 0)) ||
+        (((void)0, gSaveContext.save.time) >= (CLOCK_TIME(6, 0) + 0x10))) {
+        gSaveContext.skyboxTime = gSaveContext.save.time;
+    }
+
+    time = ((void)0, gSaveContext.save.time);
+
+    if ((time >= CLOCK_TIME(18, 0)) || (time < CLOCK_TIME(6, 0))) {
+        gSaveContext.save.isNight = true;
+    } else {
+        gSaveContext.save.isNight = false;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_UpdateNextDayTime.s")
+void func_800F6CEC(PlayState* play, u8 arg1, AdjLightSettings* adjLightSettings, EnvLightSettings* lightSettings) {
+    s32 phi_t1;
+    s32 temp_v1_2;
+    s32 temp_v1 = (arg1 % 4);
+
+    if ((((void)0, gSaveContext.save.day) >= 2) && (arg1 >= 4) && (arg1 < 8)) {
+        temp_v1_2 = (((void)0, gSaveContext.save.day) * 4) + 4;
```

Comment:
```suggestion
        temp_v1_2 = (((void)0, gSaveContext.save.day) + 1) * 4;
```
Pretty nit picky, but I do like this a little better.